ingest-app/src/cmr/ingest/api/bulk.clj

```diff
@@ -166,3 +187,31 @@
         :status-message (:status-message task-status)
         :request-json-body (:request-json-body task-status)
         :collection-statuses collection-statuses}))))
+
+(defn get-granule-task-status
+  "Get the status for the given task for the provider including collection statuses"
+  [task-id request]
+  (let [{:keys [headers request-context]} request
+        _ (validate-granule-bulk-update-result-format headers)
+        gran-task (data-gran-bulk-update/get-granule-task-by-id request-context task-id)
+        provider-id (:provider-id gran-task)]
+    (when-not provider-id
```

Comment:
should this be `when-not gran-task`?

Reply:
I want to be more specific in checking that the gran-task is something that has required values rather than an empty list or structure with errors.